### DISCUSSION TITLE
Reviving Lagom based on Pekko

### DISCUSSION THREAD

**Discussion by flexshape-laurynas** [apache/pekko#1310] [General]
Dear Pekko community,

In the old Akka days, we had this decent Scala-based opinionated framework for micro-services, called [Lagom](https://www.lagomframework.com/). It offered neat abstractions over Akka, providing end-to-end development environment for Scala-based micro-services. Unfortunately, this framework is now approaching EOL by July 1, 2024, and I'm curious whether there'd be any interest in reviving this framework in the context of Pekko (and Play 3). 

Please see my [post over here](https://github.com/lagom/lagom/issues/3366) for details.

Please let us know, whether there's some interest in this.

Kind Regards,
Laurynas

**Comment by pjfanning** [comment]
Adding this reddit discussion for some background and discussion - https://www.reddit.com/r/scala/comments/19cssqr/approaches_to_lagom_framework_migration/

From an ASF perspective, it will be difficult for us to take over Lagom because we would need the contributors and probably the copyright holder (Lightbend) to donate it to the ASF. With Akka, the BSL change was taken was the reason that we were able to proceed with the Pekko fork. Lagom license has not changed.

Maybe some Pekko contributors might be interested in working on a Lagom fork but I think it would be best to do it outside the ASF.

My understanding is that the reason that Lagom has been abandoned is because most of its core features are now in Akka and Pekko.

I would suggest that it might be best to start with some Lagom code where there are no easy alternatives and to migrate a small amount to use Pekko. This is just my opinion but I don't really see the point in migrating the whole thing.
